utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -37,39 +38,62 @@ import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream
 import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
 
 private val UNCOMPRESSED_EXTENSIONS = listOf(".pom")
-private val TAR_EXTENSIONS = listOf(".gem", ".tar", ".tar.gz", ".tgz", ".tar.bz2", ".tbz2")
+private val TAR_EXTENSIONS = listOf(".gem", ".tar")
+private val TAR_BZIP2_EXTENSIONS = listOf(".tar.bz2", ".tbz2")
+private val TAR_GZIP_EXTENSIONS = listOf(".tar.gz", ".tgz")
 private val ZIP_EXTENSIONS = listOf(".aar", ".egg", ".jar", ".war", ".whl", ".zip")
 private val SEVENZIP_EXTENSIONS = listOf(".7z")
 
-val ARCHIVE_EXTENSIONS = TAR_EXTENSIONS + ZIP_EXTENSIONS + SEVENZIP_EXTENSIONS
+val ARCHIVE_EXTENSIONS = TAR_EXTENSIONS + TAR_BZIP2_EXTENSIONS + TAR_GZIP_EXTENSIONS + ZIP_EXTENSIONS +
+        SEVENZIP_EXTENSIONS
 
+/**
+ * Unpack the [File] to [targetDirectory].
+ */
 fun File.unpack(targetDirectory: File) {
     val lowerName = name.toLowerCase()
+    if (SEVENZIP_EXTENSIONS.any { lowerName.endsWith(it) }) {
+        unpack7Zip(targetDirectory)
+    } else {
+        inputStream().unpack(name, targetDirectory)
+    }
+}
+
+/**
+ * Unpack the [InputStream] to [targetDirectory]. The compression scheme is guessed from the [fileName].
+ */
+fun InputStream.unpack(fileName: String, targetDirectory: File) {
```

Comment:
Passing a file name to a function of the stream class seems like a code smell.
What do you think about passing an enum (archive type) instead?

Reply:
I agree passing the whole file name is a bit odd here, but at least it's documented. I wanted to avoid the overhead of creating an enum because the caller then would need to have a logic to determine the enum from the file name, so that seems a bit clumsy, too. Also, just expecting the extension as a string here does not really work well for double-extensions like ".tar.gz" as ` File.extension`  would only return `gz` in that case, again making it inconvenient for the caller to pass the right information.

Reply:
Agree, thanks.